src/RealComponents.ml

```diff
@@ -0,0 +1,133 @@
+    to_string = (fun [@warning "-8"] [a ; b] -> "(- " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun _ -> [])
+  }
+]
+
+let scaling = [
+  {
+    name = "real-mult";
+    codomain = Type.REAL;
+    domain = Type.[REAL; REAL];
+    is_argument_valid = Value.(function
+                       | [x ; y]
+                         -> (x =/= Const (Real 0.)) && (x =/= Const (Real 1.)) && (x =/= Const (Real (-1.)))
+                         && (y =/= Const (Real 0.)) && (y =/= Const (Real 1.)) && (x =/= Const (Real (-1.)))
+                       | _ -> false);
+    evaluate = Value.(fun [@warning "-8"] [v1 ; v2] -> Real ((value_of v1) *. (value_of v2)));
+    to_string = (fun [@warning "-8"] [a ; b] -> "(* " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun _ -> [])
+  } ;
+  {
+    name = "real-div";
+    codomain = Type.REAL;
+    domain = Type.[REAL; REAL];
+    is_argument_valid = Value.(function
+                       | [x ; y] -> x =/= y
+                                 && (x =/= Const (Real 0.)) && (x =/= Const (Real 1.)) && (x =/= Const (Real (-1.)))
+                                 && (y =/= Const (Real 0.)) && (y =/= Const (Real 1.)) && (y =/= Const (Real (-1.)))
+                       | _ -> false);
+    evaluate = Value.(fun [@warning "-8"] [v1 ; v2] -> Real ((value_of v1) /. (value_of v2)));
+    to_string = (fun [@warning "-8"] [a ; b] -> "(/ " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun [@warning "-8"] [_ ; b] -> ["(not (= 0 " ^ b ^ "))"]);
+  }
+]
+
+let conditionals = [
+  {
+    name = "real-eq";
+    codomain = Type.BOOL;
+    domain = Type.[REAL; REAL];
+    is_argument_valid = (function
+                 | [x ; y] -> (x =/= y) && (not (is_constant x && is_constant y))
+                 | _ -> false);
+    evaluate = Value.(fun [@warning "-8"] [v1 ; v2] -> Bool Float.(equal (value_of v1) (value_of v2)));
+    to_string = (fun [@warning "-8"] [a ; b] -> "(= " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun _ -> [])
+  } ;
+  {
+    name = "real-geq";
+    codomain = Type.BOOL;
+    domain = Type.[REAL; REAL];
+    is_argument_valid = (function
+                 | [x ; y] -> (x =/= y) && (not (is_constant x && is_constant y))
+                 | _ -> false);
+    evaluate = Value.(fun [@warning "-8"] [v1 ; v2] -> Bool Float.((value_of v1) >= (value_of v2)));
+    to_string = (fun [@warning "-8"] [a ; b] -> "(>= " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun _ -> [])
+  } ;
+  {
+    name = "real-leq";
+    codomain = Type.BOOL;
+    domain = Type.[REAL; REAL];
+    is_argument_valid = (function
+                 | [x ; y] -> (x =/= y) && (not (is_constant x && is_constant y))
+                 | _ -> false);
+    evaluate = Value.(fun [@warning "-8"] [v1 ; v2] -> Bool Float.((value_of v1) <= (value_of v2)));
+    to_string = (fun [@warning "-8"] [a ; b] -> "(<= " ^ a ^ " " ^ b ^ ")");
+    global_constraints = (fun _ -> [])
+  } ;
+  {
+    name = "real-ite";
```

Comment:
Why build in a special form of if-then-else as a real component?  Isn't this already expressible in some other way?

Reply:
I am pretty sure I included this because I saw it in Saswat's num type that I was referring to, but looking back I agree that it is not necessary and probably was specific to what he needed it for. I will remove that, thank you!